relay-server/src/actors/envelopes.rs

```diff
@@ -331,34 +334,29 @@ impl EnvelopeManagerService {
         let SendMetrics {
             buckets,
             scoping,
-            partition_key,
+            extraction_mode,
         } = message;
 
-        let upstream = self.config.upstream_descriptor();
-        let dsn = PartialDsn {
-            scheme: upstream.scheme(),
-            public_key: scoping.project_key,
-            host: upstream.host().to_owned(),
-            port: upstream.port(),
-            path: "".to_owned(),
-            project_id: Some(scoping.project_id),
-        };
-
-        let mut item = Item::new(ItemType::MetricBuckets);
-        item.set_payload(ContentType::Json, serde_json::to_vec(&buckets).unwrap());
-        let mut envelope = Envelope::from_request(None, RequestMeta::outbound(dsn));
-        envelope.add_item(item);
+        #[allow(unused_mut)]
+        let mut partitions = self.config.metrics_partitions();
+        #[allow(unused_mut)]
+        let mut max_batch_size_bytes = self.config.metrics_max_batch_size_bytes();
```

Comment:
im wondering if we really need this function, couldn't we send directly to the envelopeprocessor from the projectcache?

Reply:
Kind of, it currently exists to specify different processing configs depending on the upstream sink (Kafka, Http etc.) and this is the only service that should know what the upstream is (because it is also the service which decides where to send envelopes to).

E.g. we want smaller batches for Kafka and bigger batches for Http.

Reply:
couldn't we just do that with the processing_enabled() flag in config? that sets the store to `Some` [here](https://github.com/getsentry/relay/blob/2757887550a59a9e6f64c00a55932a6cfe505a77/relay-server/src/service.rs#L310-L312) which deterministically sets the store_forwarded to `Some` on `EnvelopeManager` [here](https://github.com/getsentry/relay/blob/2757887550a59a9e6f64c00a55932a6cfe505a77/relay-server/src/service.rs#L162-L166). Although if you think it might be too implicit then i could agree with that,

Reply:
Yes we could in practice, but I specifically didn't want this implicit behaviour.

Reply:
understandable. Although maybe in the future we can look at it again, I think there's a bit too much 'ping-pong' between services in relay in general.

Reply:
I actually changed this now, thanks for bringing this up!